docs/api.rst

```diff
@@ -493,6 +494,106 @@ overlay neutral mutations on this tree sequence.
 
 .. autofunction:: msprime.mutate
 
+++++++++++++++++++++++
+Mutation Matrix Models
+++++++++++++++++++++++
+
+Mutation matrix models are specified by three things: an alphabet,
+a root distribution, and a transition matrix.
+These leave one free parameter: an overall mutation rate,
+specified by the mutation ``rate`` in the call to :func:`.mutate`.
+Concisely,
+the underlying model of mutation is a continuous-time Markov chain on the alphabet,
+started by a draw from ``root_distribution``, and
+with instantaneous transition rate from ``i`` to ``j`` that is equal to
+``rate`` multiplied by ``transition_matrix[i,j]``.
+The ``root distribution`` and every row in the ``transition_matrix``
+must give *probabilities*, i.e., they must be nonnegative numbers summing to 1.
+
+To interpret the implications,
+it helps to know how the underlying mutational process is implemented.
+First, "possible" mutations are placed on the tree,
+with a mean density equal to the ``rate``, per unit of time and sequence length.
+If ``discrete=False`` then this is an infinite-sites model,
```

Comment:
Something that just occured to  me here - it is *possible* that someone could want a continuous position but not guaranteed infinite sites behaviour. I.e., it does happen that we get multiple hits at sites when we have continuous positions, but we rejection sample them to guarantee infinite sites. Discrete and infinite sites are not quite the opposite of each other here. I guess it's even possible that someone would want infinite sites *and* discrete (if they have a large genome and low mutation rate).

Are we making a mistake by conflating discrete and infinite sites into the same parameter? It would be easy to separate them.

Reply:
Hm, good thought? But no, I think we've made the right decision. If someone really wants to simulate a discrete-sites model with 2^24 sites and the likelihood of multiple hits determined by the vagaries of pseudorandom number generation, then they can just set `sequence_length=2**24`.

Reply:
Agreed.